auth.php

```diff
@@ -395,7 +395,7 @@ public function modifyUser($user, $changes) {
 
         if(isset($changes['mail'])) {
             $found = $this->getUserByEmail($changes['mail']);
-            if($found != $user) {
+            if($found != $user && $found) {
```

Comment:
This comparison should probably be the other way round. First check that it found something, then compare it.